### DISCUSSION TITLE
Data bridge disconnections not resumed

### DISCUSSION THREAD

**Discussion by alexdelprete** [emqx/emqx#9162] [Q&A]
Hi,

I'm using EMQX v5.0.8 in my homelab with a lot of IoT devices/sensor, really happy about how it's working.

I created a Data Bridge (MQTT Sink) to export a topic to the public instance of EMQX (broker.emqx.io), and it is working fine, thanks to the support of @lafirest. It's working fine, but I noticed that sometimes the Data Bridge gets disconnected, and it doesn't auto-reconnect. The only way to resume the connection is to disable/enable the Data Bridge, so the connection is reestablished.

Here you can see that it went down and probably it reconnected automatically:

![image](https://user-images.githubusercontent.com/7027842/196064206-9832b19b-8240-4508-aca3-7530bf0c24ed.png)

Here you can see the disconnection was detected, but then I found out after a day that it wasn't working, and I had to reconnect it manually disabling/enabling:

![image](https://user-images.githubusercontent.com/7027842/196064335-41b8a23b-6658-4da7-8c69-904b92e2bb79.png)

My questions are:

1. How can I be notified if this specific event happens so that I can react manually reconnecting it?
2. What can I configure to make the bridge more resilient? This is my current configuration of the bridge:

![image](https://user-images.githubusercontent.com/7027842/196064393-a7b46aa1-1736-4e96-a00e-bd8df18dbc04.png)

Thanks a lot for the support and keep up the great work on EMQX.

**Comment by lafirest** [comment]
@alexdelprete 
Thanks for your input, this doesn't look right, we'll investigate this

**Reply by alexdelprete** [reply]
one thing: I was sending a topic with a lot of subtopics to `broker.emqx.io`, I thought maybe it was too much, so I picked only the data I actually needed to replicate and since then the bridge connection has been up. Maybe it's just a coincidence, but I thought maybe the public EMQX instance was disconnecting because of the quantity and frequency of the data, a sort of throttling.

thank you for the support. looking forward to your findings, please let me know.

**Reply by alexdelprete** [reply]
@lafirest it happened again...it has been down 9 days and I had no notice. :(

is there a way to be notified in any way on these events?

![image](https://user-images.githubusercontent.com/7027842/198902379-a8575a58-2e13-40cf-88f6-5b32786e9796.png)

**Reply by lafirest** [reply]
> @lafirest it happened again...it has been down 9 days and I had no notice. :(
> 
> is there a way to be notified in any way on these events?
> 
> ![image](https://user-images.githubusercontent.com/7027842/198902379-a8575a58-2e13-40cf-88f6-5b32786e9796.png)

Sorry, there doesn't seem to has a such event, but this will be sure logged

**Reply by lafirest** [reply]
@alexdelprete 
Sorry for the late reply, but after I investigated, I can't reproduce this problem, could you pls provide a detailed log, thanks

**Reply by alexdelprete** [reply]
can you tell me where do I configure the log level or where do I find this detailed one? thanks.

anyway, since I limited the number of messages sent to the bridge, it hasn't happened anymore. but if it happens, I'll send you the log.

also, I still can't find a method to be notified if the bridge connection goes down and doesn't restart.
